web-ng/src/app/components/map/map.component.ts

```diff
@@ -83,56 +89,68 @@ export class MapComponent implements OnInit, AfterViewInit {
   }
 
   ngAfterViewInit() {
-    this.features$.subscribe(features => {
-      this.clearGoogleMapDataLayer();
-      features.forEach(feature => {
-        if (feature instanceof GeoJsonFeature) {
-          const addedFeatures = this.map.data.addGeoJson(
-            (feature as GeoJsonFeature).geoJson
-          );
-          addedFeatures.forEach(f => {
-            f.setProperty('layerId', feature.layerId);
-          });
-        }
-      });
-    });
-
-    this.activeProject$.subscribe(project =>
-      this.map.data.setStyle(feature => {
-        const layerId = feature.getProperty('layerId');
-        const color = project.layers.get(layerId)?.color;
-        return {
-          fillColor: color,
-        };
+    this.subscription.add(
+      this.features$.subscribe(features => {
```

Comment:
Extracting these three handlers into their own methods might be more readable since it'll separate subscription/binding from the handling logic. Carefully named methods like `onFeaturesUpdate()`, `onProjectActivation` and `onEditModeChange` may also make the code more self-describing.